File: src/main/java/com/kmarzecki/communicator/repository/BoardRepository.java
package com.kmarzecki.communicator.repository;

import com.kmarzecki.communicator.model.BoardEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface BoardRepository extends JpaRepository<BoardEntity, UUID> {
    boolean existsByNameAndUsers_Username(String name, String username);
    List<BoardEntity> findAllByUsers_Username(String username);
    void deleteAlLByIdAndUsers_Username(String id, String username);
}

File: src/main/kotlin/com/kmarzecki/trollo/repository/BoardRepository.kt
package com.kmarzecki.trollo.repository

import com.kmarzecki.trollo.model.BoardEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface BoardRepository : JpaRepository<BoardEntity?, UUID?> {
    fun existsByNameAndUsers_Username(name: String, users_username: String): Boolean
    fun findAllByUsers_Username(users_username: String): List<BoardEntity>
    fun deleteAlLByIdAndUsers_Username(id: UUID, users_username: String)
    fun existsByIdAndUsers_Username(id: UUID, users_username: String): Boolean
}